internal/controllers/machine/machine_controller.go

```diff
@@ -112,8 +112,8 @@ type Reconciler struct {
 }
 
 func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
-	if r.Client == nil || r.APIReader == nil || r.ClusterCache == nil || r.RemoteConditionsGracePeriod == time.Duration(0) {
-		return errors.New("Client, APIReader and ClusterCache must not be nil and RemoteConditionsGracePeriod must not be 0")
+	if r.Client == nil || r.APIReader == nil || r.ClusterCache == nil || r.RemoteConditionsGracePeriod < 2*time.Minute {
```

Comment:
Same for KCP

Reply:
I'll add a comment. The only reason why SetupWithManager is public is because of the idea that someone can build their own binary with our controllers. So I do think for that use case it is important to make sure the input is valid. (as they won't run through our validation in main.go)